### PR TITLE
513 health bars for simplemobs!

### PR DESCRIPTION
**Pull request by tralezab**
## [About The Pull Request (video!)](https://youtu.be/BzkTNm2FJ34)

see video for demonstration. will fix the improper dead status coming too early

## Why It's Good For The Game

much prettier!

## Changelog
:cl:
add: new health hud for simple critters!
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
perfection

File: code/_onclick/hud/living.dm [hunk 1]
```diff
@@ -11,6 +11,12 @@
 	pull_icon.hud = src
 	static_inventory += pull_icon
 
-	healths = new /obj/screen/healths/living()
-	healths.hud = src
-	infodisplay += healths
+
+	//explanation: since living mobs have no limbs, they only need one of these to show health nicely.
+	//while health doll is cooler, it does not work on huge mobs and so they use the basic version.
+	if(owner.mob_biotypes & MOB_EPIC)
```
Comment (tralezab) [inline code/_onclick/hud/living.dm]: perhaps a better way to do this, may revisit

File: code/_onclick/hud/map_popups.dm [hunk 2]
```diff
@@ -6,6 +6,10 @@
 	var/list/screen_info = list()//x,x pix, y, y pix || x,y
 	var/del_on_map_removal = TRUE//this could probably be changed to be a proc, for conditional removal. for now, this works.
 
+/obj/screen/Initialize(mapload, _hud)
```
Comment (tralezab) [inline code/_onclick/hud/map_popups.dm]: remove this, unneeded

**Review by tralezab — COMMENTED** [review]
some things for me to think about

**Comment by tralezab** [comment]
im gonna play league of legends (tm) then come back and fix the linter

**Comment by Couls** [comment]
I love you

File: code/_onclick/hud/living.dm (hunk 1 above)
Comment (AnturK) [reply code/_onclick/hud/living.dm]: You can just check icon size directly.

**Comment by AnturK** [comment]
@spookydonut Something wrong with filter arg detection here ?

File: code/_onclick/hud/living.dm (hunk 1 above)
Comment (PKPenguin321) [reply code/_onclick/hud/living.dm]: One of those medical + signs would be good as a fallback

**Comment by tralezab** [comment]
Anturk it's definitely mad for all the wrong reasons. I dont need the mobicon var even, I just did it to please the linter and it's still unhappy

File: code/_onclick/hud/living.dm (hunk 1 above)
Comment (tralezab) [reply code/_onclick/hud/living.dm]: kk.

**Comment by MrPerson** [comment]
Linter looks like a false alarm which if not fixed before merging will cause everything to fail.

File: code/_onclick/hud/living.dm (hunk 1 above)
Comment (AnturK) [reply code/_onclick/hud/living.dm]: You could probably make these single type or subtype.

**Comment by AnturK** [comment]
Didn't notice before but you're changing some unrelated dmi here

**Comment by spookydonut** [comment]
> 
> 
> @spookydonut Something wrong with filter arg detection here ?

@AnturK in spacemansuite 1.2 https://github.com/SpaceManiac/SpacemanDMM/blob/17ba8291f4cff8396e5878d910fca351d1937f14/src/dreammaker/builtins.rs#L357
It's since been updated but will give false positives until then.

There's also since been this added: https://github.com/SpaceManiac/SpacemanDMM/pull/154

**Comment by spookydonut** [comment]
Also @SpaceManiac hotfix release?

**Comment by Ty-the-Smonk** [comment]
how does this look on megafauna

**Comment by tralezab** [comment]
it doesn't use it for megafauna

**Comment by SpaceManiac** [comment]
@spookydonut Pondering whether an immediate release is warranted or if we can safely UNLINT this call without forgetting to re-lint it later

**Comment by cacogen** [comment]
science

**Comment by tralezab** [comment]
@SpaceManiac I'll just DELINT for now and we'll come back to this later.

**Comment by Fikou** [comment]
image

**Comment by tralezab** [comment]
i, tralezab, humbly submit, a toast to fikou for successfully giving me specially made megafauna health bars
image

**Comment by SpaceManiac** [comment]
> I'll just DELINT for now and we'll come back to this later.

I'm okay with this, I just don't want it to be forgotten

**Comment by tralezab** [comment]
i'll keep it in mind for you

**Pull request merged**